packages/components/collapsible-panel/src/collapsible-panel.spec.js

```diff
@@ -160,85 +160,62 @@ it('should forward data-attributes', () => {
   const renderPanel = (props) => {
     const { container } = render(
-      <CollapsiblePanel
-        id="example"
-        header="Header"
-        onToggle={jest.fn()}
-        {...props}
-      >
+      <CollapsiblePanel header="Header" onToggle={jest.fn()} {...props}>
         Children
       </CollapsiblePanel>
     );
-    const getPanelHeader = () => screen.getByRole('button');
 
-    const panelContentId = CollapsiblePanel.getPanelContentId(
-      props.id || 'example'
+    const panelHeader = screen.getByRole('button');
+    const panelHeaderId = panelHeader.getAttribute('id');
+    const panelContent = container.querySelector(
+      `[aria-labelledby="${panelHeaderId}"]`
     );
-    const getPanelContent = () =>
-      container.querySelector(`[id=${panelContentId}]`);
 
     return {
-      getPanelHeader,
-      getPanelContent,
+      panelHeader,
+      panelContent,
     };
   };
   it('should have a valid aria-controls correspondence', () => {
-    const { getPanelHeader, getPanelContent } = renderPanel({
+    const { panelHeader, panelContent } = renderPanel({
       id: 'test-id',
       isClosed: false,
     });
 
-    const panelContentId = CollapsiblePanel.getPanelContentId('test-id');
-
-    // assert that the header button has the aria attribute
-    const headerButton = getPanelHeader();
-    expect(headerButton).toHaveAttribute('aria-controls', panelContentId);
-
-    // find the correspondent panel content
-    expect(getPanelContent()).toBeInTheDocument();
+    // assert that the header button has the aria attribute linked to the panel content
+    expect(panelHeader).toHaveAttribute(
+      'aria-controls',
+      panelContent.getAttribute('id')
+    );
   });
   describe('header', () => {
     it('should have aria-expanded true when panel is open', () => {
-      const { getPanelHeader } = renderPanel({ isClosed: false });
+      const { panelHeader } = renderPanel({ isClosed: false });
 
-      const headerButton = getPanelHeader();
-      expect(headerButton).toHaveAttribute('aria-expanded', 'true');
+      expect(panelHeader).toHaveAttribute('aria-expanded', 'true');
     });
 
     it('should have aria-expanded false when panel is closed', () => {
-      const { getPanelHeader } = renderPanel({ isClosed: true });
+      const { panelHeader } = renderPanel({ isClosed: true });
 
-      const headerButton = getPanelHeader();
-      expect(headerButton).toHaveAttribute('aria-expanded', 'false');
+      expect(panelHeader).toHaveAttribute('aria-expanded', 'false');
     });
   });
 
   describe('content', () => {
     it('should have aria-hidden true when panel is closed', () => {
-      const { getPanelContent } = renderPanel({ isClosed: true });
-
-      const panelContent = getPanelContent();
+      const { panelContent } = renderPanel({ isClosed: true });
 
-      expect(panelContent).toHaveAttribute('aria-hidden', 'true');
+      expect(panelContent).not.toHaveAttribute('hidden');
```

Comment:
Can you explain the difference here?

Maybe I'm missing some context around how `aria-hidden` and `hidden` work 🤔

Reply:
The `hidden` attribute is more semantically correct in this case as we want the element not to be visible for the browser nor the screen readers.
`aria-hidden` attribute only "hides" the element to screen readers which it's ok for this case since we are also hiding the element with CSS.
(_`hidden` attribute also implies `aria-hidden`_).

I wanted to follow the reference I found in the [w3c example](https://w3c.github.io/aria-practices/examples/accordion/accordion.html).
I guess [this](https://stackoverflow.com/a/49807034) is a more detailed explanation about both attributes.

Reply:
Thanks, I get it.

However, I'm still confused why before we were checking that the `aria-hidden` was set to `true` and now checking that is NOT `hidden` 🧐